docs/tutorials/go/build-an-ecommerce-app/index.md

```diff
@@ -13,14 +13,690 @@ last_update:
 
-In this four-part series, you'll explore how you can use Temporal to develop an eCommerce application with Temporal Workflows. You'll write the basic application, send reminder emails, write tests, and develop a REST API.
+### Prerequisites
 
-* [Build an eCommerce App With Temporal and Go, Part 1: Getting Started](./part1.md)
-* [Build an eCommerce App With Temporal and Go, Part 2: Reminder Emails](./part2.md)
-* [Build an eCommerce App With Temporal and Go, Part 3: Testing](./part3.md)
-* [Build an eCommerce App With Temporal and Go, Part 4: REST API](./part4.md)
+You can build the project by following this tutorial, or just grab the ready-to-go source from its [GitHub repo](https://github.com/temporalio/temporal-ecommerce).
+This repository has the full source code and can serve as a guide or a fallback if you encounter issues when working through this tutorial.
+If you want to play first and explore later, you can come back and read through the how-to and background.
 
+If you're new to Temporal, follow ["Set up a local development environment"](/getting_started/go/dev_environment/) so you're ready to build Temporal applications with Go.
+Ensure a local Temporal Service is running (e.g. with [the Temporal CLI](https://github.com/temporalio/cli)) and that you can access the Temporal Web UI from port `8233`.
+These services are necessary for you to build and run this project.
+
+## Create the Shopping cart Workflow
+
+In a typical web app, a user's shopping cart would be stored as a row or document in a database.
+While you can store shopping carts in a separate database using Temporal, you have another option: you can represent a shopping cart as a [long-lived Workflow](https://temporal.io/blog/very-long-running-workflows). This way, you get the benefits of state tracking without needing to sync from external storage.
+
+In the Go SDK, a Temporal Workflow is a function that takes 2 parameters: a Temporal Workflow context `ctx` and an arbitrary `value`.
+Temporal can handle pausing and restarting the Workflow as needed -- a Workflow is a durable construct.
+Begin by mocking out a `workflow.go` with `CartItem` and `CartState` structs, and a main Workflow function called `CartWorkflow`.
+
+```go
+package app
+
+import (
+	"go.temporal.io/sdk/workflow"
+)
+
+type (
+	CartItem struct {
+		ProductId int
+		Quantity  int
+	}
+
+	CartState struct {
+		Items []CartItem
+		Email string
+	}
+)
+
+func CartWorkflow(ctx workflow.Context, state CartState) error {
+	logger := workflow.GetLogger(ctx)
+}
+```
+
+This will be your starting point. Next, you'll add some features to the `CartWorkflow`.
+
+### Add Signal Handling for your cart
+
+To support adding and removing elements from the cart, the Workflow needs to respond to different types of [Signals](https://docs.temporal.io/encyclopedia/workflow-message-passing).
+Signals are a way to notify Workflows of external events.
```

Comment:
I think we can benefit with a bit more elaborate explanation of Signals. Does the Workflow wait for the Signal? Can you only deliver it to open Workflows etc.

Reply:
Agreed -- this wasn't really presented as a Signal tutorial originally, but I think I want to expand this explanation and add a little diagram here because that's the most useful part.